### DISCUSSION TITLE
Cannot set StelCore.currentTimeZone via RemoteControl API

### DISCUSSION THREAD

**Discussion by DrRichardDaystrom** [Stellarium/stellarium#3914] [Q&A] [answered]
Greetings!  I am trying to set the StelCore.currentTimeZone property from a Python script using the RemoteControl plugin API.

I would expect the following code, using stelproperty/set, to do the trick but it does not work ...

```python
import requests

# specify URL to Stellarium RemoteControl API
url_stellarium = 'http://localhost:8090/api/'

# set time zone ... this DOESN'T work
params_tz = {'id':'StelCore.currentTimeZone', 'value':'LTST'}
response = requests.post(url_stellarium + 'stelproperty/set')

```

Does anyone know what I am doing wrong?  The response.status_code is 200, indicating that it worked, but the time zone does not change.

I have figured out a workaround using scripts/direct ...

```python
import requests

# specify URL to Stellarium RemoteControl API
url_stellarium = 'http://localhost:8090/api/'

# set time zone ... this DOES work
params_direct = {'code': 'core.setTimezone(\"LTST\")'}
response = requests.post(url_stellarium + 'scripts/direct', data = params_direct)

```

While I am happy that I have a working method, I feel like I am not understanding how to use stelproperty/set correctly.

Any insight as to how to do this the "right" way would be sincerely appreciated.

**Comment by 10110111** [comment, marked as answer]
Where do you pass `params_tz`? In your code it's only defined but not passed.

**Reply by DrRichardDaystrom** [reply]
Thank you, kind stranger - that was exactly the problem.  I wasn't passing params_tz in the POST request.

For those following along at home, here is the corrected code using stelproperty/set ...

```
import requests

# specify URL to Stellarium RemoteControl API
url_stellarium = 'http://localhost:8090/api/'

# set time zone ... this DOESN'T work
params_tz = {'id':'StelCore.currentTimeZone', 'value':'LTST'}
response = requests.post(url_stellarium + 'stelproperty/set', data = params_tz)
```

Thanks again for finding my bug.  It seems stupid in retrospect but most bugs do.
